importer/src/main/java/org/hiero/mirror/importer/migration/BackfillAndDeduplicateBalanceMigration.java

```diff
@@ -128,28 +127,21 @@ select coalesce(max(consensus_timestamp), 0)
     private static final String SELECT_LAST_CONSENSUS_TIMESTAMP_SQL =
             "select balance from account_balance_old where consensus_timestamp = -1 and account_id = -1";
 
-    private final JdbcTemplate jdbcTemplate;
-
-    private final TimePartitionService timePartitionService;
+    private final ObjectProvider<TimePartitionService> timePartitionServiceProvider;
 
     @Getter(lazy = true)
     private final TransactionOperations transactionOperations = transactionOperations();
 
     private Long lastConsensusTimestamp;
     private long minFrequency;
 
-    @Lazy
     public BackfillAndDeduplicateBalanceMigration(
             DBProperties dbProperties,
             ImporterProperties importerProperties,
-            @Owner JdbcTemplate jdbcTemplate,
-            TimePartitionService timePartitionService) {
-        super(
-                importerProperties.getMigration(),
-                new NamedParameterJdbcTemplate(jdbcTemplate),
-                dbProperties.getSchema());
-        this.jdbcTemplate = jdbcTemplate;
-        this.timePartitionService = timePartitionService;
+            @Owner ObjectProvider<JdbcTemplate> jdbcTemplateProvider,
```

Comment:
Same comment for every other sub-class

```suggestion
            @Owner ObjectProvider<JdbcOperations> jdbcOperationsProvider,
```